### PR TITLE
APPS-2549 Bulk describe optimization in scatter-collect to cover files in arrays, structs

### PR DESCRIPTION
**Pull request by r-i-v-a**
Would depend on dxScala change 
https://github.com/dnanexus/dxScala/pull/129/files

### CODE REVIEW

File: executorCommon/src/main/scala/dx/executor/JobMeta.scala [hunk 1]
```diff
@@ -414,7 +418,6 @@ abstract class JobMeta(val workerPaths: DxWorkerPaths,
     if (queryFiles.isEmpty) {
       queryFiles
     } else {
-      logger.trace(s"Bulk describing ${queryFiles.size} files")
```
Comment (r-i-v-a) [inline executorCommon/src/main/scala/dx/executor/JobMeta.scala:417]: This logging is moved to dxApi (other PR).

File: executorCommon/src/main/scala/dx/executor/WorkflowExecutor.scala [hunk 2]
```diff
@@ -577,19 +577,15 @@ abstract class WorkflowExecutor[B <: Block[B]](jobMeta: JobMeta, separateOutputs
...
         case (accu, outputVals) => accu ++ outputVals.getOrElse(Map.empty).values.toVector
-      } map { jsv =>
-        try dxApi.dataObjectFromJson(jsv)
-        catch {
-          case _: AppInternalException => None
-        }
-      } collect {
-        case dxFile: DxFile => dxFile
+      } flatMap {
+        dxApi.flattenDxFileObjectsFromJson
```
Comment (r-i-v-a) [inline executorCommon/src/main/scala/dx/executor/WorkflowExecutor.scala:584]: The part using the new implementation from dxApi

**Review by Gvaihir — COMMENTED** [review]
Looks good! In the ticket please:
1. Run a WF prior to changes, and provide a splunk stats showing file/describe from dxScala agent. Should be some.
2. Run after changes applied showing splunk stats for the same WF (new analysis) and decreased/eliminated dxScala-made file/describe.

**Comment by r-i-v-a** [comment]
> Looks good! In the ticket please:
> 
> 1. Run a WF prior to changes, and provide a splunk stats showing file/describe from dxScala agent. Should be some.
> 2. Run after changes applied showing splunk stats for the same WF (new analysis) and decreased/eliminated dxScala-made file/describe.

done -- APPS-2549

**Review by Gvaihir — APPROVED** [review]

**Pull request merged**